lib/exhal/link.ex

```diff
@@ -34,9 +37,37 @@ defmodule ExHal.Link do
   """
   def target_url(a_link, vars \\ %{}) do
     case a_link do
-      %__MODULE__{templated: true} -> {:ok, UriTemplate.expand(a_link.target_url, vars)}
-      %__MODULE__{target_url: nil} -> :error
-      _                      -> {:ok , a_link.target_url}
+      %__MODULE__{href: nil} ->
```

Comment:
It's perfectly fine to do `%{href: nil}`, etc... unless you are adamant about checking the type.  If you are trying to guard against the wrong type, I would put a guard before the case statement.